pytest_redis/factories/proc.py

```
@@ -76,6 +83,35 @@ def redis_proc_fixture(
+        n = 0
+        used_ports: set[int] = set()
+        while True:
+            try:
+                redis_port = _redis_port(port, config, used_ports)
+                port_filename_path = port_path / f"redis-{redis_port}.port"
+                if redis_port in used_ports:
+                    raise PortForException(
+                        f"Port {redis_port} already in use, "
+                        f"probably by other instances of the test. "
+                        f"{port_filename_path} is already used."
+                    )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)